src/track/macroaction.h

```diff
@@ -0,0 +1,47 @@
+#pragma once
+
+#include "audio/frame.h"
+#include "engine/engine.h"
+#include "proto/macro.pb.h"
+namespace proto = mixxx::track::io;
+
+/// A MacroAction is the building block of a Macro.
+/// It contains a position as well as the action to be taken at that position.
+///
+/// Note that currently only jumps to a target position are available,
+/// but that is subject to change.
+class MacroAction {
+  public:
+    enum class Type : uint32_t {
+        Jump = 0
+    };
```

Comment:
Should this be registered with ` Q_ENUM` to ensure support within Qt metatype?

Reply:
This would require a lot of overhead, as this class is not QObject based and there is no moc_ file generation needed otherwise. I think this should therefore not done, until somebody has a real need for it.